src/road_map.cpp

```diff
@@ -40,13 +40,13 @@ RoadBits GetAnyRoadBits(TileIndex tile, RoadTramType rtt, bool straight_tunnel_b
 				default:
 				case ROAD_TILE_NORMAL:   return GetRoadBits(tile, rtt);
 				case ROAD_TILE_CROSSING: return GetCrossingRoadBits(tile);
-				case ROAD_TILE_DEPOT:    return DiagDirToRoadBits(GetRoadDepotDirection(tile));
+				case ROAD_TILE_DEPOT:    return DiagDirToStraightRoadBits(GetRoadDepotDirection(tile));
```

Comment:
`GetAnyRoadBits` is used in various places to check for multiple things. Some of them are functions like  `ScriptRoad::AreRoadTilesConnected`. Any place that calls this function without limiting the tile type might get a different result now. E.g. for an AI, a road would suddenly be connected to the back of a depot.

Reply:
I'm not sure how this should be dealt with. In-game, it sure *looks* like road depots and bay stops have 2 road bits judging by the number of vehicles that fit on their tiles, and the infrastructure counter is incremented by 2 when placing either of them; so it would make sense to go with 2 road tiles per bay stop/depot across the board.

Reply:
As for an AI thinking that the back of a road depot is connected to adjacent road tiles, this doesn't seem all that different to an AI thinking that the back of a *rail depot* is connected to adjacent *rails*; and clearly this doesn't happen. So perhaps the fix used for that is applicable to road depots and bay stops as well.

Reply:
Of course two road bits is the correct amount for the infrastructure count, but it is _not_ the correct value for `GetAnyRoadBits`. Road bits are half-tile bits, rail track bits are full-tile, thus internally simply different.

Don't just modify `GetAnyRoadBits`, modify the conversion command where the value ends up at.

Reply:
Alright, so should I add a constant multiplier like ROAD_DEPOT_TRACKBIT_FACTOR, analogously to TUNNELBRIDGE_TRACKBIT_FACTOR, in `economy_type.h`?

Reply:
If I do that there would also need to be a special clause in the conversion logic for depots and bay stops specifically, which is kind of ugly but should work

Reply:
I've amended the changes

Reply:
Looks good.
Can I bother you to use the two new constants instead of the hardcoded 2 in `CmdBuildRoadDepot`, `RemoveRoadDepot` (both road_cmd.cpp), `CmdBuildRoadStop` and `RemoveRoadStop` (both station_cmd.cpp)?

Reply:
Sure

Reply:
looking at those commands now, I think it makes more sense to change ROAD_BAY_STOP_TRACKBIT_FACTOR to just ROAD_STOP_TRACKBIT_FACTOR and use it for both drive-through and bay stops, I hope that's fine.

Reply:
Alright, that ought to have done it.
I apologize for getting a bit caught up on the details, I'm fairly new to all of this. Looking at the code, though, I think it would be nice to homogenise the code a bit since at times the same data seems to be extracted in a bunch of different ways, which sometimes don't actually agree with each other.